source/scpd/tests/VTableTest.d

```diff
@@ -73,3 +78,44 @@ unittest
     assert(n.__vptr[2] is &BonTestB.vfunc2);
     assert(n.__vptr[3] is &BonTestB.vfunc1);
 }
+
+/// It checks the offset of virtual on C++ and D for any class.
+unittest
+{
+    import std.format;
+    import std.string;
+
+    long[string] offset_d;
+    long val_d;
+    string offsets;
+
+    static foreach (e; VTableCheckClasses)
+    {
+        offset_d.clear();
+        val_d = 0;
+        offsets = "";
+
+        mixin(
+        q{
+            static foreach (member; __traits(allMembers, %1$s))
+            {
+                mixin(
+                q{
+                    static if (__traits(isVirtualMethod, %1$s.%2$s) && (`%2$s` != `__xdtor`))
+                    {
+                        static if (`%2$s` == `__dtor`)
+                            val_d += 2;
```

Comment:
Why += 2?

Reply:
The destructor always occupies two pointers on the vtable.
So I added two.

Reply:
Ah yes, I think on the D side there are two destructors. One is `__dtor` and one is `__xdtor`.

Reply:
The offsets of TestA are 
TestA._dtor : 0
TestA.vfunc1 : 1
TestA.vfunc2 : 2
TestA.__xdtor : 3
If A and B come out one after another, we can add 1 more, but since that's not the case, we have to add 2.

Reply:
``` C++
class A
{
public:
    virtual ~A() {};
    virtual void vfunc1(){};
    virtual void vfunc2(){};
};

class B
{
public:
    virtual void vfunc1(){};
    virtual void vfunc2(){};
    virtual ~B() {};
};

int CPPTest1 ()
{
    auto pAf1 = &A::vfunc1;
    printf("A.vfunc1 : %d\n", (long)reinterpret_cast<void *&>(pAf1) / sizeof(long));
    auto pAf2 = &A::vfunc2;
    printf("A.vfunc2 : %d\n", (long)reinterpret_cast<void *&>(pAf2) / sizeof(long));
    auto pBf1 = &B::vfunc1;
    printf("B.vfunc1 : %d\n", (long)reinterpret_cast<void *&>(pBf1) / sizeof(long));
    auto pBf2 = &B::vfunc2;
    printf("B.vfunc2 : %d\n", (long)reinterpret_cast<void *&>(pBf2) / sizeof(long));
}
```
Result is 
A.vfunc1 : 2
A.vfunc2 : 3
B.vfunc1 : 0
B.vfunc2 : 1

In A, 0 and 1 are destructors, and in B, 2 and 3 are destructors.